packages/squiggle-lang/src/value/index.ts

```diff
@@ -26,12 +37,29 @@ If you add a new value class, don't forget to add it to the "Value" union type b
 "vBlah" functions are just for the sake of brevity, so that we don't have to prefix any value creation with "new".
 */
 
-class VArray {
+class VArray implements Indexable {
   readonly type = "Array" as const;
   constructor(public value: Value[]) {}
   toString(): string {
     return "[" + this.value.map((v) => v.toString()).join(",") + "]";
   }
+
+  get(key: Value) {
+    if (key.type === "Number") {
+      const index = key.value | 0; // TODO - fail on non-integer indices?
+      if (index >= 0 && index < this.value.length) {
+        return this.value[index];
```

Comment:
Might be obvious - but won't `` this.value[index]`` return an error if out of bounds, that you could use? Instead of redoing that check?

Reply:
No, out of bounds index lookup on arrays returns `undefined` in JS. It would probably fail later but with a cryptic error.